Calculator/Calculator/Controller/ViewController.swift

```diff
@@ -6,13 +6,169 @@
+        stack.addArrangedSubview(operands)
+        
+        containerStackView.addArrangedSubview(stack)
+        scrollView.setContentOffset(CGPoint(x: 0, y: scrollView.contentSize.height - scrollView.bounds.height), animated: true)
+        scrollView.layoutIfNeeded()
+    }
+ 
+    @IBAction func calculateButtonDidTap(_ sender: UIButton) {
+           guard currentLabelText.isEmpty == false,
+                 let operatorSign = OperatorLabel.text  else { return }
+           inputList.removeLast()
+           inputList.append(operatorSign)
+           
+           updateScrollView()
+           inputList.append(currentLabelText)
+           
+           let calculate = inputList.reduce("") { current, next in current + " " + next}
+           
+           var formula = ExpressionParser.parse(from: calculate)
+           guard let result = formula.result(),
+                 result.isNaN == false  else {
+               OperandsLabel.text = Double.nan.description
+               isDotClicked = false
+               
+               inputList.removeAll()
+               currentLabelText.removeAll()
+               OperatorLabel.text = ""
+               return
+           }
+         
+           let intResult = Int(result)
+                
+           inputList.removeAll()
+           currentLabelText.removeAll()
+           OperatorLabel.text = ""
+           
+           if result == Double(intResult) {
+               OperandsLabel.text = formatNumber(intResult.description)
+               isDotClicked = false
+           } else {
+               OperandsLabel.text = formatNumber(result.description)
+               isDotClicked = true
+           }
+       }
+    
+    @IBAction func allClearButtonDidTap(_ sender: UIButton) {
+        inputList.removeAll()
+        currentLabelText.removeAll()
+        OperatorLabel.text?.removeAll()
+        OperandsLabel.text?.removeAll()
+        containerStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+    }
 
+    @IBAction func clearCurrentOperands(_ sender: UIButton) {
+        currentLabelText.removeAll()
+        OperandsLabel.text = "0"
+    }
+    
+    @IBAction func changeNumberSign(_ sender: UIButton) {
+        guard var operands = OperandsLabel.text,
+              operands != "0"  else { return }
+        
+        if operands.first == "-" {
+            operands.removeFirst()
+        } else {
+            operands = "-" + operands
+        }
+        currentLabelText = operands
+        OperandsLabel.text = operands
+    }
+    
+    @IBAction func dotButtonDidTap(_ sender: UIButton) {
+        if currentLabelText.isEmpty {
+            currentLabelText = "0."
+            OperandsLabel.text = currentLabelText
+        } else if isDotClicked == false {
+            currentLabelText += "."
+            OperandsLabel.text = currentLabelText
+            isDotClicked = true
+        } else if isDotClicked == true {
+            return
+        }
```

Comment:
첫번째 조건과 2,3 번째 조건은 다르다고 생각하는데 else if 로 연결해주신 이유가 있을까요? 
```suggestion
        guard currentLabelText.isEmpty == false else {
            currentLabelText = "0."
            OperandsLabel.text = currentLabelText
        }
         
        if isDotClicked == false {
            currentLabelText += "."
            OperandsLabel.text = currentLabelText
            isDotClicked = true
        } else if isDotClicked == true {
            return
        }
```

이렇게 조건을 구분해주면 좋을 것 같아요

Reply:
상황에 따라 세가지중 하나의 결과만 있을거라 생각하고 다른 조건임에도 한번에 이어서 쓴 것 같습니다! 수정했습니다!